### ISSUE TITLE
Release Definition Visual Studio Test task v2.* does not find test assemblies

### ISSUE THREAD

**Issue by Koshak118** [Microsoft/vsts-tasks#6680]
## Description
When trying to run Visual Studio Test task v. 2.* within some Release Definition (after the deployment task) it fails  - test assemblies not found. 
```

2018-03-11T13:45:58.9299389Z ##[section]Starting: VsTest - testAssemblies
2018-03-11T13:45:58.9308588Z ==============================================================================
2018-03-11T13:45:58.9308898Z Task         : Visual Studio Test
2018-03-11T13:45:58.9309304Z Description  : Run unit and functional tests (Selenium, Appium, Coded UI test, etc.) using the Visual Studio Test runner. Test frameworks that have a Visual Studio test adapter such as xUnit, NUnit, Chutzpah, etc. can also be run. Tests can be distributed on multiple agents using this task (version 2).
2018-03-11T13:45:58.9309987Z Version      : 2.3.22
2018-03-11T13:45:58.9310218Z Author       : Microsoft Corporation
2018-03-11T13:45:58.9310499Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=835764)
2018-03-11T13:45:58.9310806Z ==============================================================================
2018-03-11T13:46:00.2276627Z Run the tests locally using vstest.console.exe
2018-03-11T13:46:00.2277087Z ========================================================
2018-03-11T13:46:00.2277775Z Test selector : Test assemblies
2018-03-11T13:46:00.2278155Z Test assemblies : **\*test*.dll,!**\*TestAdapter.dll,!**\obj\**
2018-03-11T13:46:00.2278744Z Test filter criteria : TestCategory=GUITests
2018-03-11T13:46:00.2279050Z Search folder : D:\a\r1\a
2018-03-11T13:46:00.2279394Z Run settings file : D:\a\r1\a
2018-03-11T13:46:00.2279964Z Run in parallel : false
2018-03-11T13:46:00.2280285Z Run in isolation : false
2018-03-11T13:46:00.2281981Z Path to custom adapters : null
2018-03-11T13:46:00.2282309Z Other console options : null
2018-03-11T13:46:00.2282694Z Code coverage enabled : false
2018-03-11T13:46:00.2283486Z Rerun failed tests: false
2018-03-11T13:46:00.2283876Z VisualStudio version selected for test execution : 14.0
2018-03-11T13:46:00.3538282Z ========================================================
2018-03-11T13:46:00.3606879Z ##[warning]No test assemblies found matching the pattern: **\*test*.dll,!**\*TestAdapter.dll,!**\obj\**.
2018-03-11T13:46:00.4897315Z ##[section]Finishing: VsTest - testAssemblies
```
The same task in a Build Definition runs just fine, thus tried to add a path to custom test assemblies from that successful Build Definition task **(F:\VSTSBuild\workspace\8\s\VS\$PROJECTNAME\bin\Release)** but it does not help:

```
2018-03-11T11:24:14.1592030Z ##[section]Starting: VsTest - testAssemblies
2018-03-11T11:24:14.1599679Z ==============================================================================
2018-03-11T11:24:14.1600003Z Task         : Visual Studio Test
2018-03-11T11:24:14.1600437Z Description  : Run unit and functional tests (Selenium, Appium, Coded UI test, etc.) using the Visual Studio Test runner. Test frameworks that have a Visual Studio test adapter such as xUnit, NUnit, Chutzpah, etc. can also be run. Tests can be distributed on multiple agents using this task (version 2).
2018-03-11T11:24:14.1600997Z Version      : 2.3.22
2018-03-11T11:24:14.1601205Z Author       : Microsoft Corporation
2018-03-11T11:24:14.1601467Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=835764)
2018-03-11T11:24:14.1601755Z ==============================================================================
2018-03-11T11:24:14.8054390Z Run the tests locally using vstest.console.exe
2018-03-11T11:24:14.8054796Z ========================================================
2018-03-11T11:24:14.8055468Z Test selector : Test assemblies
2018-03-11T11:24:14.8055818Z Test assemblies : **\*test*.dll,!**\*TestAdapter.dll,!**\obj\**
2018-03-11T11:24:14.8056150Z Test filter criteria : TestCategory=GUITests
2018-03-11T11:24:14.8056455Z Search folder : D:\a\r1\a
2018-03-11T11:24:14.8056858Z Run settings file : D:\a\r1\a
2018-03-11T11:24:14.8058062Z Run in parallel : false
2018-03-11T11:24:14.8058411Z Run in isolation : false
2018-03-11T11:24:15.0560015Z ##[error]Error: Path to custom adapters '**F:\VSTSBuild\workspace\8\s\VS\$PROJECTNAME\bin\Release**' should be a directory and it should exist.
2018-03-11T11:24:15.1786711Z ##[section]Finishing: VsTest - testAssemblies
```

The '**F:\VSTSBuild\workspace\8\s\VS\$PROJECTNAME\bin\Release**' path taken from the logs of the Build Definition test task which runs properly.

It looks like **"D:\a\r1\a"** is the root directory and the path to the custom assembles is calculated related to the root directory - because '**F:\VSTSBuild\workspace\8\s\VS\$PROJECTNAME\bin\Release**' does exist on file system.

Is that possible to set the root directory in some way? Because considering even drive letters are different I cannot calculate the relative path properly.

## Steps to reproduce
Create Visual Studio Test task v. 2.* within some Build Definition Agent Phase with criteria **(TestCategory!=GUITests)** and default assemblies names **(**\$(BuildConfiguration)\*test*.dll;-:**\obj\**)**
Create Visual Studio Test task v. 2.* within some Release Definition Agent Phase *with criteria 
**(TestCategory=GUITests)** and default values for test assemblies **(**\*test*.dl, !**\*TestAdapter.dll, !**\obj\**)**
Make Build Definition run Release Definition
Run Build Definition

## Expected behavior
Build Definition tests run
Release definition test run

## Actual behavior
Build Definition tests run
Release definition test fails while not finding the test assemblies

## Environment
VSTS 2017
Release Definition Visual Studio Test task v. 2.*

Account, project and definition names can be provided upon request

**Comment by prawalagarwal** [comment]
@Koshak118 Can you run your release with release variable "system.debug=true" and provide us the logs at devops_tools[at]microsoft[dot]com. Thanks.

**Comment by ShreyasRmsft** [comment]
Hi @Koshak118, 

**Couple of things:**

- Path to custom adapters is not the same as path to test assemblies. This input field is something else. It is used to provide test adapters required to run tests of different frameworks.

- You will have to change the **Search folder** input to point explicitly point to the test dlls.

- Also the way you are going about it seems wrong. Your build definition is supposed to upload the test dlls as artifacts. And your release definition is supposed to download them into the artifacts folder and then proceed ahead. If you are not uploading the test dlls in the build definition then I suggest you run the tests in the build definition itself.

- I can take a look at your build and release definitions if you want to know the right way to go about it. You can attach screenshots here. It would also help if you can tell me what exactly is your requirement with the build definition and release definition and where you want to run your tests and why.

**Comment by Koshak118** [comment]
Hey Shreys

Thanks for the answer
So it seems there is a fundamental issue with how I think things should
work

Can we do an interactive session to solve the problem, like Skype session
with screen sharing?

**Comment by ShreyasRmsft** [comment]
@Koshak118 we can. Can you share your skype handle with me? Either here or to shrer@microsoft.com.
I will initiate a call.

**Comment by Koshak118** [comment]
Can we schedule it in 4 hours from now please?

**Comment by ShreyasRmsft** [comment]
@Koshak118 That is outside my working hours (I'm available from 11am to 5pm IST). 

We can correspond through mails also. Screen shots should help. Or you can give me (shrer@microsoft.com) permissions to a clone of your definition and I can take a look and give suggestions.

**Comment by Koshak118** [comment]
OK, so I'll ping you when I get back to the office and we'll see how we
proceed

Thanks for the fast responses meanwhile

**Comment by Koshak118** [comment]
Well, I tried to add you as a project contributor but with no success
Anyway, here are the 2 screenshots: the first of the working build definition, which was pretty much straightforward 

![workingbuilddefinition](https://user-images.githubusercontent.com/6772460/37598700-9ecbfa5e-2b8b-11e8-90a7-459e570586cd.PNG)

and the second is for the broken release definition (after removing custom paths)
![notwokringreleasedefinition](https://user-images.githubusercontent.com/6772460/37598764-c228108c-2b8b-11e8-97c7-b70776d6abf4.PNG)

They looks pretty much the same for a reason  - I'd like to run the same assemblies but with different configuration - (TestCategory!=GUITests) for the first and (TestCategory=GUITests) for the latter, which are literally Selenium GUI tests which should run after the Azure deployment task

**Comment by ShreyasRmsft** [comment]
@Koshak118 You'll need to add me to the AAD before adding me to the vsts account.

**Comment by ShreyasRmsft** [comment]
Yeah so you're going about it the wrong way. Follow how to create a [build definition from here](https://docs.microsoft.com/en-us/vsts/build-release/actions/ci-cd-part-1#create-a-build-definition) and how to create a [linked release definition from here ](https://docs.microsoft.com/en-us/vsts/build-release/actions/ci-cd-part-1#create-a-release-definition)

**Comment by ShreyasRmsft** [comment]
Your build definition should upload the built binaries and your release definition is meant to download those uploaded binaries via artifacts.

**Comment by ShreyasRmsft** [comment]
If you are not doing this then there is no point in having a separate release definition as you clearly expect the steps in the release definition to run on the same machine where the build ran since you are expecting the test dlls to be present on the machine. You might as well have the steps you are performing in the release definition in the build definition itself.

**Comment by ShreyasRmsft** [comment]
But if you are bent on having both then just use the search folder input to hardcode the path to the built test dlls. But this is clearly not the recommended way to go about this and it will most definitely not scale.

**Comment by Koshak118** [comment]
Ok, I'll take a look on it shortly. The motivation of doing it as I did is to run some integration tests locally, then upload the binaries to some some test Azure app and then run the some Selenium tests - from the same assemblies

**Comment by ShreyasRmsft** [comment]
Yes then you will need to run the integration tests in your build definition and then publish these binaries as build artifacts and link your release definition to your build and consume these artifacts there.

**Comment by Koshak118** [comment]
Speaking of **Search Folder** - that what I was meant by "changing the root path" . Didin't find it in "Variables" section of the definition though, where it is hiding?

**Comment by Koshak118** [comment]
Regarding the linking build and release definitions - so yeah, there is a "After Release" trigger which starts the release definition which defines "Artifacts filter" to be artifacts from that specific build definition which doing its testing job well

Should I define something else?

**Comment by ShreyasRmsft** [comment]
you should add a task to upload those artifacts. just check the link on creating a build definition or use the default template for building .Net apps. ideally you shouldn't have to change the search folder input. You can just type in whatever value you want. It doesn't have to be a build variable. Please take a look at some sample templates. Also please take a look at other blog posts regarding the vstest task usage as well as for creating proper build definitions and release definitions

**Comment by ShreyasRmsft** [comment]
@Koshak118 once you add the task to upload the build artifacts and get your release to download them just set the Search folder to $(System.ArtifactsDirectory) and things should work.

**Comment by Koshak118** [comment]
Well so the build definition does have the "publish artifacts" task at the last one which looks like that:

![publishartifacts](https://user-images.githubusercontent.com/6772460/37661320-35793756-2c5d-11e8-847d-8e18b60020b0.PNG)

So there are 2 question right now:

1. How to make the release definition to download those artifacts or the v2.* does it automatically and nothing should be done?
2. Where is that "Search Folder" variable which should be set?

**Comment by ShreyasRmsft** [comment]
@Koshak118 Search Folder is an input. Please take a look at the task and read the input descriptions and the blogs. Everything is clearly documented there. 

How to make the release definition to download those artifacts or the v2.* does it automatically: This is also documented please search for it online. When you create a release definition if you can read the on screen messages and take a look at the UI you will find out all this on your own. 

I am closing this issue now. The task is intuitive enough to use, please report back if you get any actual issues.

**Comment by Koshak118** [comment]
Guys, if things were so straightforward and intuitive I wouldn't open this issue.

So right, "Search Folder" is an input field of the task, but when I set it to **"F:\VSTSBuild\workspace\8\s\VS\Tests\bin\Release"** things end up with error
 `"##[error]Error: Search folder: 'F:\VSTSBuild\workspace\8\s\VS\Tests\bin\Release' should be a directory and it should exist."`  - despite the fact that it is an existing folder

So I set the "Search Field" to " $(System.ArtifactsDirectory)" , added the task of unzipping the artifacts after downloading of them and that what I am getting

Artifacts downloaded:

```
2018-03-21T11:24:50.8817225Z ##[section]Starting: Download artifact - Rule Them All
2018-03-21T11:24:50.9016987Z ==============================================================================
2018-03-21T11:24:50.9017278Z Task         : Download Build Artifacts
2018-03-21T11:24:50.9017506Z Description  : Download Build Artifacts
2018-03-21T11:24:50.9017705Z Version      : 0.131.1
2018-03-21T11:24:50.9017920Z Author       : Microsoft Corporation
2018-03-21T11:24:50.9018426Z Help         : 
2018-03-21T11:24:50.9018697Z ==============================================================================
2018-03-21T11:24:51.9973046Z Linked artifacts count:  1
2018-03-21T11:24:51.9982405Z Downloading artifacts from: https://say2eat.visualstudio.com//_apis/resources/Containers/1335576?itemPath=drop&isShallow=true&api-version=4.1-preview.4
2018-03-21T11:24:52.2623781Z Downloading drop/Say2eat.API.deploy-readme.txt to D:\a\r1\a\Rule Them All\drop\Say2eat.API.deploy-readme.txt
2018-03-21T11:24:52.2659074Z Downloaded drop/Say2eat.API.deploy-readme.txt to D:\a\r1\a\Rule Them All\drop\Say2eat.API.deploy-readme.txt
2018-03-21T11:24:52.3186680Z Downloading drop/Say2eat.API.deploy.cmd to D:\a\r1\a\Rule Them All\drop\Say2eat.API.deploy.cmd
2018-03-21T11:24:52.3198933Z Downloaded drop/Say2eat.API.deploy.cmd to D:\a\r1\a\Rule Them All\drop\Say2eat.API.deploy.cmd
2018-03-21T11:24:52.3907509Z Downloading drop/Say2eat.API.SetParameters.xml to D:\a\r1\a\Rule Them All\drop\Say2eat.API.SetParameters.xml
2018-03-21T11:24:52.3926635Z Downloaded drop/Say2eat.API.SetParameters.xml to D:\a\r1\a\Rule Them All\drop\Say2eat.API.SetParameters.xml
2018-03-21T11:24:52.4584799Z Downloading drop/Say2eat.API.SourceManifest.xml to D:\a\r1\a\Rule Them All\drop\Say2eat.API.SourceManifest.xml
2018-03-21T11:24:52.4585209Z Downloaded drop/Say2eat.API.SourceManifest.xml to D:\a\r1\a\Rule Them All\drop\Say2eat.API.SourceManifest.xml
2018-03-21T11:24:52.6287773Z Downloading drop/Say2eat.API.zip to D:\a\r1\a\Rule Them All\drop\Say2eat.API.zip
2018-03-21T11:24:53.7194635Z Downloading drop/Say2eat.BotFramework.SourceManifest.xml to D:\a\r1\a\Rule Them All\drop\Say2eat.BotFramework.SourceManifest.xml
2018-03-21T11:24:53.7198522Z Downloaded drop/Say2eat.BotFramework.SourceManifest.xml to D:\a\r1\a\Rule Them All\drop\Say2eat.BotFramework.SourceManifest.xml
2018-03-21T11:24:53.7199943Z Downloading drop/Say2eat.BotFramework.SetParameters.xml to D:\a\r1\a\Rule Them All\drop\Say2eat.BotFramework.SetParameters.xml
2018-03-21T11:24:53.7200630Z Downloaded drop/Say2eat.BotFramework.SetParameters.xml to D:\a\r1\a\Rule Them All\drop\Say2eat.BotFramework.SetParameters.xml
2018-03-21T11:24:53.7201858Z Downloading drop/Say2eat.BotFramework.deploy.cmd to D:\a\r1\a\Rule Them All\drop\Say2eat.BotFramework.deploy.cmd
2018-03-21T11:24:53.7203279Z Downloaded drop/Say2eat.BotFramework.deploy.cmd to D:\a\r1\a\Rule Them All\drop\Say2eat.BotFramework.deploy.cmd
2018-03-21T11:24:53.7208179Z Downloading drop/Say2eat.BotFramework.deploy-readme.txt to D:\a\r1\a\Rule Them All\drop\Say2eat.BotFramework.deploy-readme.txt
2018-03-21T11:24:53.7211257Z Downloaded drop/Say2eat.BotFramework.deploy-readme.txt to D:\a\r1\a\Rule Them All\drop\Say2eat.BotFramework.deploy-readme.txt
2018-03-21T11:24:53.7214534Z Downloading drop/Say2eat.BotFramework.zip to D:\a\r1\a\Rule Them All\drop\Say2eat.BotFramework.zip
2018-03-21T11:24:53.7214942Z Downloaded drop/Say2eat.API.zip to D:\a\r1\a\Rule Them All\drop\Say2eat.API.zip
2018-03-21T11:24:53.7221062Z Downloaded drop/Say2eat.BotFramework.zip to D:\a\r1\a\Rule Them All\drop\Say2eat.BotFramework.zip
2018-03-21T11:24:54.4104278Z Total Files: 10, Processed: 10, Skipped: 0, Failed: 0, Download time: 2.41 secs, Download size: 26.456MB
2018-03-21T11:24:54.5625900Z Successfully downloaded artifacts to D:\a\r1\a/Rule Them All
2018-03-21T11:24:54.5653087Z ##[section]Finishing: Download artifact - Rule Them All
```

Artifacts unzipped (by another task)

```
2018-03-21T11:29:04.2066782Z ##[section]Starting: Unzip artifacts
2018-03-21T11:29:04.2069292Z ==============================================================================
2018-03-21T11:29:04.2069556Z Task         : Unzip file
2018-03-21T11:29:04.2069793Z Description  : extracts files from a zip, a pattern can be applied to extract only parts of the file.
2018-03-21T11:29:04.2070039Z Version      : 0.9.5
2018-03-21T11:29:04.2070223Z Author       : Yod Labs
2018-03-21T11:29:04.2070420Z Help         : 
2018-03-21T11:29:04.2070749Z ==============================================================================
2018-03-21T11:29:04.4239792Z (node:2888) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
2018-03-21T11:29:04.4398671Z unzipping D:\a\r1\a\Rule Them All\drop\Say2eat.BotFramework.zip using pattern  to folder D:\a\r1\a
2018-03-21T11:29:05.0995516Z ##[section]Finishing: Unzip artifacts
```

No assemblies found:
```

2018-03-21T11:29:05.1033908Z ##[section]Starting: VsTest - testAssemblies
2018-03-21T11:29:05.1039447Z ==============================================================================
2018-03-21T11:29:05.1039720Z Task         : Visual Studio Test
2018-03-21T11:29:05.1040171Z Description  : Run unit and functional tests (Selenium, Appium, Coded UI test, etc.) using the Visual Studio Test runner. Test frameworks that have a Visual Studio test adapter such as xUnit, NUnit, Chutzpah, etc. can also be run. Tests can be distributed on multiple agents using this task (version 2).
2018-03-21T11:29:05.1040569Z Version      : 2.3.27
2018-03-21T11:29:05.1040795Z Author       : Microsoft Corporation
2018-03-21T11:29:05.1041238Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=835764)
2018-03-21T11:29:05.1041532Z ==============================================================================
2018-03-21T11:29:05.7511862Z Run the tests locally using vstest.console.exe
2018-03-21T11:29:05.7512177Z ========================================================
2018-03-21T11:29:05.7512642Z Test selector : Test assemblies
2018-03-21T11:29:05.7513006Z Test assemblies : **\*test*.dll,!**\*TestAdapter.dll,!**\obj\**
2018-03-21T11:29:05.7513358Z Test filter criteria : TestCategory=GUITests
2018-03-21T11:29:05.7513692Z Search folder : D:\a\r1\a
2018-03-21T11:29:05.7514019Z Run settings file : D:\a\r1\a
2018-03-21T11:29:05.7514444Z Run in parallel : false
2018-03-21T11:29:05.7514759Z Run in isolation : false
2018-03-21T11:29:05.7516401Z Path to custom adapters : null
2018-03-21T11:29:05.7516725Z Other console options : null
2018-03-21T11:29:05.7517302Z Code coverage enabled : false
2018-03-21T11:29:05.7517987Z Rerun failed tests: false
2018-03-21T11:29:05.7518424Z VisualStudio version selected for test execution : 14.0
2018-03-21T11:29:06.2141783Z ========================================================
2018-03-21T11:29:06.2243879Z ##[warning]No test assemblies found matching the pattern: **\*test*.dll,!**\*TestAdapter.dll,!**\obj\**.
2018-03-21T11:29:06.3682176Z ##[section]Finishing: VsTest - testAssemblies
```

The weird issue that **"D:\a\r1\a"** does not even exists and I cannot see what actually is going on the file system. There are no **"a"** folder on partition **"D"** - neither system nor hidden

**Comment by bryanmacfarlane** [comment]
Re-opening based on comments

**Comment by ShreyasRmsft** [comment]
@bryanmacfarlane just for the record this issue seems more related to using the build and release flows more than using the vstest task.

**Comment by ShreyasRmsft** [comment]
@Koshak118 You don't need to add an explicit task to download the artifacts. Release does this for you. You can link an artifact to a release and it downloads and unzips them to the System.Artifacts folder.

Are you using a private agent or a Hosted agent?

Seeing as to how long this issue seems going, If you can gimme a skype call during my working hours I can **go out of my way** (Coz the issue you are facing is not an issue with the VsTest task which we own, but with just getting your test dlls on to the machine to execute) and help you set up your Release for you. If this is not convenient then add me (shrer@microsoft.com) to the AAD of you org and give me permissions to your team project and I will be more than happy to set it up for you.

**Comment by Koshak118** [comment]
@ShreyasRmsft I've added you to the organization AAD, you were supposed to get the invitation 
You are a contributor of a project now

Also a Skype credentials were sent to shrer@microsoft.com

Let's get rid of this weirdo

**Comment by ShreyasRmsft** [comment]
@Koshak118, sent an invite and waiting for acceptance.

**Comment by Koshak118** [comment]
OK, so we are all set now

The main issue was that the Release definition has been run by Hosted agent rather than our own one

@ShreyasRmsft - thanks for the assistance by Skype, great job

**Comment by arturo2019** [comment]
hello good night, I have the following problem when I integrate a project of test studio and I attach the evidence of the configuration:

![2018-07-11 21_10_58-demoselenium-final - visual studio team services](https://user-images.githubusercontent.com/2317824/42607085-1773b0e2-854f-11e8-991b-31664637e57f.png)

![2018-07-09 13_05_29-demoselenium-final - visual studio team services](https://user-images.githubusercontent.com/2317824/42607061-f76a3e88-854e-11e8-9776-e08c280f4270.png)


![2018-07-09 13_04_39-](https://user-images.githubusercontent.com/2317824/42607003-99d017a2-854e-11e8-8a12-2ff250ee43b8.png)


The release can not detect the tests that were assembled in the build

`2018-07-12T01:00:50.1666758Z DiscoveryMessage : Logging TestHost Diagnostics in file: D:\a\_temp\Slice-71_VSTest_TestPlatform.host.18-07-12_01-00-48_62511_6.diag
2018-07-12T01:00:50.7601461Z ##[debug]ProxyDiscoveryRequest: HandleDiscoveryComplete Entered
2018-07-12T01:00:50.7601949Z ##[debug]ProxyDiscoveryRequest: HandleDiscoveredTests Entered
2018-07-12T01:00:50.7615019Z ##[debug]TestCaseFilter.FilterTests : Test Filtering invoked.
2018-07-12T01:00:50.7640128Z ##[debug]TestCaseFilter.GetFilteredTestCases : Starting filtering of discovered test cases.
2018-07-12T01:00:50.7641983Z ##[debug]TestCaseFilter.FilterTests : Test Filtering completed.
2018-07-12T01:00:50.7642414Z ##[debug]ProxyDiscoveryRequest: Dispose Entered
2018-07-12T01:00:50.7642741Z ##[debug]ProxyDiscoveryRequest: Abort Entered
2018-07-12T01:00:50.7643043Z ##[debug]PERF WARNING: DiscoverTests.PerformDiscovery : CreateDiscoveryRequest: took 3042.3294 ms
2018-07-12T01:00:50.7656225Z ##[debug]PERF WARNING: DiscoverTests.PerformDiscovery : VstestUtilities.GetTestPlatform: took 16299.5239 ms
2018-07-12T01:00:50.7724727Z ##[debug]PERF WARNING: DiscoverTests.PerformDiscovery : TestPlatformClient.PublisherMefHost.Instance: took 16306.8078 ms
2018-07-12T01:00:50.7851993Z Number of testcases discovered : 0
2018-07-12T01:00:50.7868043Z ##[debug]PERF: DiscoverTests:AddTestCases: took 0.003 ms
2018-07-12T01:00:50.8040303Z Discovered tests 0 from sources
2018-07-12T01:00:50.8040488Z =================================================================
2018-07-12T01:00:50.8040776Z ##[debug]DiscoverTests.PerformDiscovery : Test Discovery completed.
2018-07-12T01:00:50.8045936Z ##[debug]PERF WARNING: DiscoverTests.PerformDiscovery: took 16957.0379 ms
2018-07-12T01:00:50.8212012Z [RunStatistics]This execution slice with id '71', received '1' testcases to execute out of which '0' is discovered.
2018-07-12T01:00:50.8212467Z ##[debug]PERF: Execution:FetchPreviousPhaseData: took 16985.0774 ms
2018-07-12T01:00:50.8235256Z ##[debug]PERF: Execution:Run: took 1.3988 ms
2018-07-12T01:00:50.8368483Z ##[debug]ExecuteAndPublish.Run : Cleanup invoked.
2018-07-12T01:00:50.8368843Z ##[debug]Cleanup.Cleanup : Creating new cleanup instance.
2018-07-12T01:00:50.8374337Z ##[debug]Cleanup.Run : Initialize cleanup
2018-07-12T01:00:50.8391239Z ##[debug]Processed: ##vso[task.uploadfile]D:\a\_temp\Slice-71_VSTest_TestPlatform.diag
2018-07-12T01:00:50.8391915Z ##[debug]Processed: ##vso[task.uploadfile]D:\a\_temp\Slice-71_VSTest_TestPlatform.host.18-07-12_01-00-48_62511_6.diag
2018-07-12T01:00:50.8396240Z ##[debug]Cleanup.Run : Cleanup finished
2018-07-12T01:00:50.8396467Z ##[debug]PERF WARNING: ExecutionAndPublish:Run: took 17121.8137 ms
2018-07-12T01:00:51.7541349Z ##[error]An exception occurred while executing the task. So the task is not executed. System.AggregateException: One or more errors occurred. ---> System.InvalidOperationException: No test assemblies found on the test machine matching the source filter criteria or no tests discovered matching test filter criteria. Verify that test assemblies are present on the machine and test filter criteria is correct.
2018-07-12T01:00:51.7552512Z ##[debug]Processed: ##vso[task.logissue type=error;]An exception occurred while executing the task. So the task is not executed. System.AggregateException: One or more errors occurred. ---> System.InvalidOperationException: No test assemblies found on the test machine matching the source filter criteria or no tests discovered matching test filter criteria. Verify that test assemblies are present on the machine and test filter criteria is correct.
2018-07-12T01:00:51.7553042Z    at Microsoft.VisualStudio.TestService.VstestAdapter.Execution.Run(ExecutionStateContext stateModdelContext, CancellationToken cancellationToken) in D:\V2.0\A2\_work\37\s\Ta\Agent\VS.TestService.VstestAdapter\Phases\Execution.cs:line 53
2018-07-12T01:00:51.7553763Z    at Microsoft.VisualStudio.TestService.VstestAdapter.ExecutionAndPublish.Run(ExecutionStateContext stateModelContext, CancellationToken cancellationToken) in D:\V2.0\A2\_work\37\s\Ta\Agent\VS.TestService.VstestAdapter\Phases\ExecuteAndPublish.cs:line 35
2018-07-12T01:00:51.7554073Z    at System.Threading.Tasks.Task`1.InnerInvoke()
2018-07-12T01:00:51.7554205Z    at System.Threading.Tasks.Task.Execute()
2018-07-12T01:00:51.7554330Z    --- End of inner exception stack trace ---
2018-07-12T01:00:51.7554649Z ---> (Inner Exception #0) System.InvalidOperationException: No test assemblies found on the test machine matching the source filter criteria or no tests discovered matching test filter criteria. Verify that test assemblies are present on the machine and test filter criteria is correct.
2018-07-12T01:00:51.7554998Z    at Microsoft.VisualStudio.TestService.VstestAdapter.Execution.Run(ExecutionStateContext stateModdelContext, CancellationToken cancellationToken) in D:\V2.0\A2\_work\37\s\Ta\Agent\VS.TestService.VstestAdapter\Phases\Execution.cs:line 53
2018-07-12T01:00:51.7555638Z    at Microsoft.VisualStudio.TestService.VstestAdapter.ExecutionAndPublish.Run(ExecutionStateContext stateModelContext, CancellationToken cancellationToken) in D:\V2.0\A2\_work\37\s\Ta\Agent\VS.TestService.VstestAdapter\Phases\ExecuteAndPublish.cs:line 35
2018-07-12T01:00:51.7555848Z    at System.Threading.Tasks.Task`1.InnerInvoke()
2018-07-12T01:00:51.7556103Z    at System.Threading.Tasks.Task.Execute()<---
2018-07-12T01:00:51.7556187Z 
2018-07-12T01:00:51.7556475Z ##[debug]ExecutionStateResolver.SetNextPhase : Execution State Resolver called. Curren Phase: Execution`




Please I would appreciate guidance to solve this problem

**Comment by ShreyasRmsft** [comment]
@arturo2019 please create a fresh issue. Problems like this are mostly unrelated. We will look into it. You can paste the link to the new issue here for speedy tracking.

**Comment by arturo2019** [comment]
@ShreyasRmsft this is the new https://github.com/Microsoft/vsts-tasks/issues/7708 thank you very much

**Comment by LarsPellarin** [comment]
I had a similar issue where the cause was I had copied a semi-colon separated search pattern in the "Test files" field from my `v1` Visual Studio Test task definition to my `v2` definition 
(i.e. `build/Tests/*Test.dll;-:**\obj\**`) . 
Apparently `v2` does not support semi-colon separation so i changed to new-line separation and it worked. 

Hope it will be of value to someone with this issue, otherwise let me know if i should post this somewhere else.

**Comment by kpzara** [comment]
Hi,

I am experiencing the same issue. I tried to change the pattern so it will look for the test automation dll files for Easy Repro and for the dll files I have created.

2018-12-10T05:18:40.4786270Z ##[section]Starting: VSTest
2018-12-10T05:18:40.4793309Z ==============================================================================
2018-12-10T05:18:40.4793387Z Task         : Visual Studio Test
2018-12-10T05:18:40.4793438Z Description  : Run unit and functional tests (Selenium, Appium, Coded UI test, etc.) using the Visual Studio Test (VsTest) runner. Test frameworks that have a Visual Studio test adapter such as MsTest, xUnit, NUnit, Chutzpah (for JavaScript tests using QUnit, Mocha and Jasmine), etc. can be run. Tests can be distributed on multiple agents using this task (version 2).
2018-12-10T05:18:40.4793480Z Version      : 2.143.7
2018-12-10T05:18:40.4793531Z Author       : Microsoft Corporation
2018-12-10T05:18:40.4793569Z Help         : [More information](https://go.microsoft.com/fwlink/?LinkId=835764)
2018-12-10T05:18:40.4793605Z ==============================================================================
2018-12-10T05:18:41.7705771Z SystemVssConnection exists true
2018-12-10T05:18:41.7706866Z SystemVssConnection exists true
2018-12-10T05:18:41.8062113Z Running tests using vstest.console.exe runner.
2018-12-10T05:18:41.8062205Z ======================================================
2018-12-10T05:18:41.8062454Z Test selector : Test assemblies
2018-12-10T05:18:41.8062527Z Test filter criteria : null
2018-12-10T05:18:41.8062592Z Search folder : D:\a\1\s
2018-12-10T05:18:41.8063031Z VisualStudio version selected for test execution : latest
2018-12-10T05:18:41.9496412Z Run in parallel : false
2018-12-10T05:18:41.9497179Z Run in isolation : false
2018-12-10T05:18:41.9497462Z Path to custom adapters : null
2018-12-10T05:18:41.9498264Z Other console options : null
2018-12-10T05:18:41.9498520Z Code coverage enabled : false
2018-12-10T05:18:41.9498650Z Diagnostics enabled : false
2018-12-10T05:18:41.9503571Z SystemVssConnection exists true
2018-12-10T05:18:41.9533569Z Run the tests locally using vstest.console.exe
2018-12-10T05:18:41.9533672Z ========================================================
2018-12-10T05:18:41.9544588Z Test selector : Test assemblies
2018-12-10T05:18:41.9549084Z Test assemblies : **\Release\**\*UIAutomation*.dll**\obj**
2018-12-10T05:18:41.9549396Z Test filter criteria : null
2018-12-10T05:18:41.9553317Z Search folder : D:\a\1\s
2018-12-10T05:18:41.9555096Z Run settings file : D:\a\1\s
2018-12-10T05:18:41.9556942Z Run in parallel : false
2018-12-10T05:18:41.9558969Z Run in isolation : false
2018-12-10T05:18:41.9581736Z Path to custom adapters : null
2018-12-10T05:18:41.9582167Z Other console options : null
2018-12-10T05:18:41.9584097Z Code coverage enabled : false
2018-12-10T05:18:41.9585216Z Diagnostics enabled : false
2018-12-10T05:18:41.9590023Z Rerun failed tests: false
2018-12-10T05:18:41.9593102Z VisualStudio version selected for test execution : latest
2018-12-10T05:18:42.8699816Z ========================================================
2018-12-10T05:18:42.9759033Z ##[warning]No test assemblies found matching the pattern: **\Release\**\*UIAutomation*.dll**\obj**.
2018-12-10T05:18:42.9759424Z ======================================================
2018-12-10T05:18:43.0135733Z ##[section]Finishing: VSTest

Changed it on the yml file but no luck. Still getting the issue.

**Comment by prawalagarwal** [comment]
@kpzara please open a new issue. This is required as we no longer track closed issues. Thanks!
